.github/workflows/preview.yaml

```diff
@@ -0,0 +1,39 @@
+name: PR Preview
+
+on: [pull_request]
+
+permissions:
+  pull-requests: write # allow surge-preview to create/update PR comments
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setupconda
+
+      - name: Build JupyterBook
+        # NOTE: login shell activates conda environment
+        shell: bash -l {0}
+        run: |
+          jupyter-book build ./ --warningiserror --keep-going
+
+      - name: Dump Build Logs
+        if: always()
+        run: |
+          if (test -a _build/html/reports/*log); then cat _build/html/reports/*log ; fi
+
+      - name: Publish to Surge
+        id: preview_step
+        uses: afc163/surge-preview@v1
+        with:
+          surge_token: ${{ secrets.SURGE_TOKEN }}
```

Comment:
@lsetiawan, i just added this to the repo secrets.

Reply:
Thank you so much! And thanks for the comments, I'll take a look and make the changes 🙏🏽